shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -287,6 +299,26 @@ - (BOOL)launchEngine {
   return YES;
 }
 
+// Macos does not call keyUp: on a key while the command key is pressed. This results in a loss
+// of a key event once the modified key is released. This method listens registers the
+// ViewController as a listener for a keyUp event before it's handled by NSApplication, and should
+// NOT modify the event to avoid any unexpected behavior.
+- (void)listenForMetaModifiedKeyUpEvents {
+  NSAssert(_keyUpMonitor == nil, @"_keyUpMonitor was already created");
+  FlutterViewController* __weak weakSelf = self;
+  _keyUpMonitor = [NSEvent
+      addLocalMonitorForEventsMatchingMask:NSEventMaskKeyUp
+                                   handler:^NSEvent*(NSEvent* event) {
+                                     // Intercept keyUp only for events triggered on the current
+                                     // view.
+                                     if (([[event window] firstResponder] == weakSelf.view) &&
```

Comment:
Please also check that `weakSelf.view` is non-nil so this can't accidentally match if there's no Flutter view and no first responder. (Likely impossible with the current registration logic, but better safe than sorry.)

Reply:
Done